phoenix-core/src/main/java/org/apache/phoenix/queryserver/tool/QueryServerCanaryTool.java

```diff
@@ -0,0 +1,476 @@
+            return results;
+        }
+
+        @Override
+        public void publishResults() throws Exception {
+
+            Gson gson = new GsonBuilder().setPrettyPrinting().create();
+            String resultJson = gson.toJson(results);
+            System.out.println(resultJson);
+        }
+
+        @Override
+        public void clearResults() {
+            results.clear();
+        }
+    }
+
+    /**
+     * Implementation of File Out Sink
+     */
+    public static class FileOutSink implements Sink {
+        private List<CanaryTestResult> results = new ArrayList<>();
+        File dir;
+        String logfileName;
+        String propFileName = "pqs-canary-file-sink.properties";
+
+        public FileOutSink() throws Exception {
+            Properties prop = new Properties();
+            InputStream input = ClassLoader.getSystemResourceAsStream(propFileName);
+            if (input == null) {
+                throw new Exception("Cannot load " + propFileName + " file for " + "FileOutSink.");
+            }
+            prop.load(input);
+            logfileName = prop.getProperty("file.name");
+            dir = new File(prop.getProperty("file.location"));
+            dir.mkdirs();
+        }
+
+        @Override
+        public void updateResults(CanaryTestResult result) {
+            results.add(result);
+        }
+
+        @Override
+        public List<CanaryTestResult> getResults() {
+            return results;
+        }
+
+        @Override
+        public void publishResults() throws Exception {
+            Gson gson = new GsonBuilder().setPrettyPrinting().create();
+            String resultJson = gson.toJson(results);
+            String fileName = logfileName + "-" + new SimpleDateFormat("yyyy.MM.dd.HH" + ".mm" +
+                    ".ss").format(new Date()) + ".log";
+            File file = new File(dir, fileName);
+            Files.write(Bytes.toBytes(resultJson), file);
+        }
+
+        @Override
+        public void clearResults() {
+            results.clear();
+        }
+    }
+
+    private static final Logger LOG = LoggerFactory.getLogger(QueryServerCanaryTool.class);
+
+    private static String getCurrentTimestamp() {
+        return new SimpleDateFormat("yyyy.MM.dd.HH.mm.ss.ms").format(new Date());
+    }
+
+    private static Namespace parseArgs(String[] args) {
+
+        ArgumentParser parser = ArgumentParsers.newFor("PQS Canary Test Tool").build()
+                .description("PQS Canary Test Tool");
+
+        parser.addArgument("--hostname", "-hn").type(String.class).nargs("?").help("Hostname on "
+                + "which PQS is running.");
+
+        parser.addArgument("--port", "-p").type(String.class).nargs("?").help("Port on which PQS " +
+                "" + "is running.");
+
+        parser.addArgument("--constring", "-cs").type(String.class).nargs("?").help("Pass an " +
+                "explicit connection String to connect to PQS. default: null");
```

Comment:
Is the default really null?  Looks like below if you don't specify this, it uses jdbc:phoenix:thin:serialization=PROTOBUF;url= ...
Maybe we should put that as an example here so the usage is clearer